packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -205,17 +189,20 @@ export class PendingStateManager implements IDisposable {
 
     /**
      * Applies stashed ops at their reference sequence number so they are ready to be ACKed or resubmitted
+     * @param seqNum - Sequence number at which to apply ops. Will apply all ops if seqNum is undefined.
      */
-    public async applyStashedOpsAt(seqNum: number) {
+    public async applyStashedOpsAt(seqNum?: number) {
         // apply stashed ops at sequence number
         while (!this.initialStates.isEmpty()) {
             // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
             const nextState = this.initialStates.peekFront()!;
             if (nextState.type === "message") {
-                if (nextState.referenceSequenceNumber > seqNum) {
-                    break; // nothing left to do at this sequence number
-                } else if (nextState.referenceSequenceNumber > 0 && nextState.referenceSequenceNumber < seqNum) {
-                    throw new Error("loaded from snapshot too recent to apply stashed ops");
+                if (seqNum !== undefined) {
+                    if (nextState.referenceSequenceNumber > seqNum) {
+                        break; // nothing left to do at this sequence number
+                    } else if (nextState.referenceSequenceNumber > 0 && nextState.referenceSequenceNumber < seqNum) {
```

Comment:
Note to self: I wonder who is using a refSeq of 0?

Reply:
I looked back at the original PR and I guess this was from tests where empty file creation was still being used instead of detached containers, so I've removed it.